libsolidity/formal/SMTChecker.cpp

```diff
@@ -72,13 +74,13 @@ bool SMTChecker::visit(FunctionDefinition const& _function)
 			"Assertion checker does not yet support constructors and functions with modifiers."
 		);
 	m_currentFunction = &_function;
-	// We only handle local variables, so we clear at the beginning of the function.
-	// If we add storage variables, those should be cleared differently.
 	m_interface->reset();
 	m_variables.clear();
+	m_variables.insert(m_stateVariables.begin(), m_stateVariables.end());
```

Comment:
`m_variables` is mostly used in a way that works for both local and state vars, that's why I decided to have only one map (instead of this one and another `m_stateVariables`) and add the state vars to `m_variables` in the beginning of a function visit